### DISCUSSION TITLE
v1.1.44 Animated AVIF, Quite OK Image support

### DISCUSSION THREAD

**Discussion by sylikc** [sylikc/jpegview#152] [General]
## 1.1.44

This is a big release with many new and changed behaviors.  Documentation will be updated later and possibly some video tutorials on how to use the drag-to-change-crop-orientation feature, but for now, read on below!

For existing users, note that `JPEGView.ini` has been reorganized.  It might be a little painful to set your changed options in the new INI, but now the options are grouped logically, and future options should be easier to find!

### New features:
- Add support for **Animated AVIF (AVIF Sequence)**
  PR #143, thanks [qbnu](https://github.com/qbnu)
- **QOI (Quite OK Image) Read/Write support**, (also supported on XP)
  PR #150, thanks [qbnu](https://github.com/qbnu)
- New Mouse Gesture: `CTRL + SHIFT + LMouse Drag` => Instantly zoom into selected area
  Modified backport feature from fork by [David Neon](https://github.com/sdneon)
- New Crop options:
  - Fixed to image aspect ratio (actual ratio displays in menu)
  - 7:5 (common print ratio)
  - 1:1 (common web size)
- New INI file setting: `FlashWindowAlert`
  When _WrapAroundFolder=false_, briefly flash window when attempting to navigate beyond the beginning or end of the file list.
  (Default to enabled, since it's visual UI feedback)
- New INI file setting: `BeepSoundAlert`
  When _WrapAroundFolder=false_, play the system beep sound when attempting to navigate beyond the beginning or end of the file list.
  (Default to disabled, since audio feedback can be an annoyance)
- When using a fixed crop mode, and resizing of crop/selection box:
  Dragging the handles in `Top-Left` and `Bottom-Right` will change to portrait or landscape orientation if dragging in the direction to "pull" the crop into the other orientation.  (Top-Left pull by w, Bottom-Right pull by h)
  (Effectively doubles the aspect ratios natively supported into portrait orientations)

### Bugs fixed:
- Fixed: Transparency color for some image types was reversed BGR
- Fixed: TGA used background color for transparency instead of transparency color
- When viewing the last frame of a multi-frame image, at the end of the file list, with option _WrapAroundFolder=false_, attempting to navigate to the next frame will no longer loop back to frame 0

### Changed behavior:
- Restores ICC Profile support for PNGs by using GDI+ when ForceGDIPlus=true or UseEmbeddedColorProfiles=true options set.  NOTE: Disables animation.
  PR #149, thanks [qbnu](https://github.com/qbnu)
- When cursor is visible (not in full screen) always zoom to cursor when executing command `IDM_TOGGLE_FIT_TO_SCREEN_100_PERCENTS` (Spacebar by default)
  Previous behavior: Always zoomed to center when a new image was loaded
- Replaced INI option `IsSortedUpcounting` with `FileSortAscending` (it is an English translatable word, and clearer on what it is related to)
- When renaming, either by clicking on the filename on the bottom panel, or a keymap to IDM_RENAME, only select the filename part without the extension part for easier renaming
- Completely unbound the zoom. Previously allowed: 10% - 1600%
  Zoom out: Unlimited to as small as possible
  Zoom in: Unlimited as long as the image "rescaled" by zoom factor is < 65535 in any dimension (internal limitation)
- Crop ratios submenu is now sorted by increasing ratios, from square (1:1) to increasingly oblong

### Other changes:
- New INI file setting: `ZoomPausePercent`
  The default behavior of JPEGview is to briefly pause at 100% when continuously zooming in or out. (This ensures you can always get to perfectly 100%)
  This new setting allows the user to configure to a different percentage, or disable to zoom continuously with no interruption
- Reorder JPEGView.ini options grouping similar options and adding comment headers
  Options should be easier to find and logically grouped. Previously, options were haphazardly added and weren't sorted in any particular order
  (It might be a little painful to move your existing options over to the new INI format, but future options will be much easier to find!)
- Updated Russian translation, thanks Dmitry Yerokhin
- Updated Japanese translation, thanks [maboroshin](https://github.com/maboroshin)
- Updated Chinese Simplified translation, thanks [dukeng](https://github.com/dunkeng)
- New strings.txt English reference for translators, it will always be current to the code
- Second part of movie/slideshow menus now translatable
- Updated to Turbo JPEG library version 2.1.91 (2023-02-08)
- Updated to JPEG XL library version 0.8.1 (2023-02-03)

### Installer Changes:
- Add all built-in supported file extensions to Window's "Open With >" submenu
- Add "Edit with JPEGView" option to Windows Explorer for all extensions JPEGView can save to
- 64-bit installs into "C:\Program Files" on new installs (existing installs will use previous install location)

## SHA-256 Checksums

Verify the release with these checksums on the release files:
```sha256sum
fa6671581362ec8e6f2c0ad8eaec35aed0d2c16cc3f79af27ddd3f39cae29258 *JPEGView32_1.1.44.7z
dd11ad28cbb01e986771193c0115cfdedccd59b7b9f0d950738f727ed2ec3e42 *JPEGView32_en-us_1.1.44.msi
28a20e15d3cecd3fb6a722b9689f3d9bad7ee5a55a50e06cf6c812e7f5839d93 *JPEGView32_WinXP_1.1.44.zip
9aada7da72312045a7fe092d2a6aa16c0c406bb668741e8af7024e624c646e8f *JPEGView64_1.1.44.7z
14284405c51fe4b9493aa5c3f85ad4efa839bd5139d52b0d55eb301bb5c20109 *JPEGView64_en-us_1.1.44.msi
ceb506dbf8be86c12e1838c1564aac5a238bf91e693ebdfe3919b80e6f20bb9a *JPEGView_1.1.44.7z
5fb99b94480b7c08df30f57df66b086d730ae4ae01cb94e154e444944bd1d1ac *JPEGView_1.1.44.zip
```

<hr /><em>This discussion was created from the release <a href='https://github.com/sylikc/jpegview/releases/tag/v1.1.44'>v1.1.44 Animated AVIF, Quite OK Image support</a>.</em>

**Comment by LaurentGrenet** [comment]
Hi Kevin,

thanks a lot for this new version.
Pls find attached the french translation file enhanced with the two missing new translations (("Same as image" and "User-defined")

Next time, if new translations are needed in a future version, don't hesitate to ask for them before new version publishing.

[strings_fr.txt](https://github.com/sylikc/jpegview/files/10848454/strings_fr.txt)

**Reply by sylikc** [reply]
Thanks.  Yeah, I totally forgot when adding the feature like two weeks ago to even create the translatable string.  Then I was doing my spot checking right before packaging and realized I forgot to call the localization routines on that specific new feature.

I'll remember to ask next time :D

**Comment by pureby** [comment]
Is it normal that this update is still not available on Winget?

**Reply by sylikc** [reply]
I usually wait a few days before posting to WinGet... in the past, after making a release, people usually either find bugs right away or it's a good solid release.

right now there's a critical bug #158 which will get a new release later today hopefully... I also was debugging #153 which I think I have a good idea how to workaround... though that might not make the next release.

**Reply by sylikc** [reply]
I'll be making another release now, and since it's not just a bugfix, the version number will change.  Then after a few more days of that I'll send that up to WinGet.

I think fixing #153 will appease a lot of users, as that's been an outstanding bug reported multiple times... I finally was able to reproduce it and step through the conditions which triggered the freeze, without having to disable AVX2 entirely... I mean it was bugging me too that I couldn't figure out when it actually happened, although there was known workarounds
